src/test/java/io/camunda/zeebe/bpmnassert/testengine/db/InMemoryZeebeDbTransactionTest.java

```diff
@@ -0,0 +1,479 @@
+
+    transaction.run(
+        () -> {
+
+          // when
+          final ZeebeDbTransaction sameTransaction = transactionContext.getCurrentTransaction();
+
+          // then
+          assertThat(transaction).isEqualTo(sameTransaction);
+        });
+  }
+
+  @Test
+  public void shouldNotReopenTransactionWithOperations() {
+    // given
+    oneKey.wrapLong(1);
+    oneValue.wrapLong(-1);
+
+    twoKey.wrapLong(52000);
+    twoValue.wrapLong(192313);
+
+    threeKey.wrapLong(Short.MAX_VALUE);
+    threeValue.wrapLong(Integer.MAX_VALUE);
+
+    transactionContext.runInTransaction(
+        () -> {
+
+          // when
+          final ZeebeDbTransaction sameTransaction = transactionContext.getCurrentTransaction();
+          sameTransaction.run(
+              () -> {
+                oneColumnFamily.put(oneKey, oneValue);
+                twoColumnFamily.put(twoKey, twoValue);
+                threeColumnFamily.put(threeKey, threeValue);
+              });
+          sameTransaction.commit();
+
+          // then it is committed but available in this transaction
+          assertThat(oneColumnFamily.exists(oneKey)).isTrue();
+          oneColumnFamily.delete(oneKey);
+
+          assertThat(twoColumnFamily.exists(twoKey)).isTrue();
+          assertThat(threeColumnFamily.exists(threeKey)).isTrue();
+        });
+
+    // then it is committed but available in this transaction
+    assertThat(oneColumnFamily.exists(oneKey)).isFalse();
+    assertThat(twoColumnFamily.exists(twoKey)).isTrue();
+    assertThat(threeColumnFamily.exists(threeKey)).isTrue();
+  }
+
+  @Test
+  public void shouldRollbackTransaction() throws Exception {
+    // given
+    oneKey.wrapLong(1);
+    oneValue.wrapLong(-1);
+
+    twoKey.wrapLong(52000);
+    twoValue.wrapLong(192313);
+
+    threeKey.wrapLong(Short.MAX_VALUE);
+    threeValue.wrapLong(Integer.MAX_VALUE);
+
+    final ZeebeDbTransaction transaction = transactionContext.getCurrentTransaction();
+    transaction.run(
+        () -> {
+          oneColumnFamily.put(oneKey, oneValue);
+          twoColumnFamily.put(twoKey, twoValue);
+          threeColumnFamily.put(threeKey, threeValue);
+        });
+
+    // when
+    transaction.rollback();
+
+    // then
+    assertThat(oneColumnFamily.exists(oneKey)).isFalse();
+    assertThat(twoColumnFamily.exists(twoKey)).isFalse();
+    assertThat(threeColumnFamily.exists(threeKey)).isFalse();
+  }
+
+  @Test
+  public void shouldGetValueInTransaction() {
+    // given
+    final AtomicLong actualValue = new AtomicLong(0);
+    oneKey.wrapLong(1);
+    oneValue.wrapLong(-1);
+
+    // when
+    transactionContext.runInTransaction(
+        () -> {
+          oneColumnFamily.put(oneKey, oneValue);
+          final DbLong value = oneColumnFamily.get(oneKey);
+          actualValue.set(value.getValue());
+        });
+
+    // then
+    assertThat(actualValue.get()).isEqualTo(-1);
+    assertThat(oneColumnFamily.get(oneKey).getValue()).isEqualTo(-1);
```

Comment:
🙃 I don't think this assertion should be part of this test. The test wants to assert that we can get values in a transaction. At this stage the transaction has been completed, so getting this value has nothing to do with getting a value in a transaction. This case is already being verified in other tests.